apps/sportmode/settings.js

```diff
@@ -0,0 +1,54 @@
+(function(back) {
+  const FILE = "sportmode.json";
+  const settings = Object.assign({
+    mode: -1,
+  }, require('Storage').readJSON(FILE, true) || {});
+
+  function writeSettings() {
+    require('Storage').writeJSON(FILE, settings);
+  }
+
+  // see Espruino/libs/misc/vc31_binary/algo.h
+  const SPORT_MODES = [
+    /*LANG*/"Normal",
+    /*LANG*/"Running",
+    /*LANG*/"Ride bike",
+    /*LANG*/"Jump rope",
+    /*LANG*/"Swimming",
+    /*LANG*/"Badminton",
+    /*LANG*/"Table tennis",
+    /*LANG*/"Tennis",
+    /*LANG*/"Climbing",
+    /*LANG*/"Walking",
+    /*LANG*/"Basketball",
+    /*LANG*/"Football",
+    /*LANG*/"Baseball",
+    /*LANG*/"Volleyball",
+    /*LANG*/"Cricket",
+    /*LANG*/"Rugby",
+    /*LANG*/"Hockey",
+    /*LANG*/"Dance",
+    /*LANG*/"Spinning",
+    /*LANG*/"Yoga",
+    /*LANG*/"Sit up",
+    /*LANG*/"Treadmill",
+    /*LANG*/"Gymnastics",
+    /*LANG*/"Boating",
+    /*LANG*/"Jumping jack",
+    /*LANG*/"Free training",
+  ];
+
+  E.showMenu({
+    "" : { "title" : /*LANG*/"HRM sport mode" },
+    "< Back" : () => back(),
+    /*LANG*/'Sport mode': {
+      value: settings.mode,
+      min: -1, max: SPORT_MODES.length-1,
+      format: v => v === -1 ? /*LANG*/"Auto" : SPORT_MODES[v],
+      onchange: v => {
+        settings.mode = v;
+        writeSettings();
```

Comment:
If a user's got fastload enabled, this won't take effect until a full reload of the watch - perhaps we should set `hrmSportMode` here too?

Reply:
Actually leaving settings does a reload

Reply:
So it does, nice!

Reply:
But I'm thinking of adding this line anyway. What if we want to open this setting from e.g. heart rate monitor.

Reply:
Yeah that's a fair point, nice catch with the remove handler too